From 1367b51e190d526d767ff863d7520bf61a0e68c2 Mon Sep 17 00:00:00 2001
From: Sergey Shelukhin <sershe@apache.org>
Date: Mon, 11 Jan 2016 13:33:06 -0800
Subject: [PATCH 1032/1164] CDH-51487 HIVE-12768 : Thread safety: binary
 sortable serde decimal deserialization (Vikram
 Dixit K, reviewed by Gunther Hagleitner, Sergey
 Shelukhin)

Change-Id: I07df36b65531bd4b9ed324fc81a34410fcade311
---
 .../serde2/binarysortable/BinarySortableSerDe.java |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/serde/src/java/org/apache/hadoop/hive/serde2/binarysortable/BinarySortableSerDe.java b/serde/src/java/org/apache/hadoop/hive/serde2/binarysortable/BinarySortableSerDe.java
index 25090bd..85ea90c 100644
--- a/serde/src/java/org/apache/hadoop/hive/serde2/binarysortable/BinarySortableSerDe.java
+++ b/serde/src/java/org/apache/hadoop/hive/serde2/binarysortable/BinarySortableSerDe.java
@@ -125,7 +125,6 @@
 
   boolean[] columnSortOrderIsDesc;
 
-  private static byte[] decimalBuffer = null;
   public static Charset decimalCharSet = Charset.forName("US-ASCII");
 
   @Override
@@ -404,9 +403,7 @@ static Object deserialize(InputByteBuffer buffer, TypeInfo type,
           length++;
         } while (true);
 
-        if(decimalBuffer == null || decimalBuffer.length < length) {
-          decimalBuffer = new byte[length];
-        }
+        final byte[] decimalBuffer = new byte[length];
 
         buffer.seek(start);
         for (int i = 0; i < length; ++i) {
-- 
1.7.9.5

